src/meta/src/barrier/mod.rs

```diff
@@ -601,13 +661,8 @@ impl GlobalBarrierManager {
             span,
         } = self.scheduled_barriers.pop_or_default().await;
 
-        let all_nodes = self
-            .context
-            .metadata_manager
-            .list_active_streaming_compute_nodes()
-            .await
-            .unwrap();
-        self.state.resolve_worker_nodes(all_nodes);
+        self.state
+            .resolve_worker_nodes(self.active_streaming_nodes.current().values().cloned());
```

Comment:
Can we maintain active streaming nodes inside state? So we can further avoid updating on it for each barrier.
If not, can we just only resolve when a change event emitted?

Reply:
Good idea! I have changed to only resolve it when a change event is emitted.